File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
from collections import Counter
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        l=[]
        c=Counter(nums)
        for n,fre in c.items():
            l.append([fre,n])
        l.sort(reverse=True,key=lambda x:x[0])
        res=[]
        for i in range(k):
            res.append(l[i][-1])
        return res

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
         unordered_map<int, int> freq;
        for(int x : nums){
            freq[x]++;
        }

        vector<pair<int,int>> v;
        for(auto &p : freq){
            v.push_back({p.second, p.first});
        }

        sort(v.begin(), v.end(), greater<pair<int,int>>());

        vector<int> res;
        for(int i = 0; i < k; i++){
            res.push_back(v[i].second);
        }

        return res;
    }
};